src/app/route/Checkout/Checkout.container.js

```diff
@@ -463,6 +472,27 @@ export class CheckoutContainer extends PureComponent {
     }
 
     async savePaymentInformation(paymentInformation) {
+        const { totals: { is_virtual } } = this.props;
+        const {
+            billing_address: {
+                firstname: billingFirstName,
+                lastname: billingLastName
+            }
+        } = paymentInformation;
+
+        /**
+         * If cart contains only virtual products then set firstname & lastname
+         * from billing step into shippingAddress for user creating.
+         */
+        if (is_virtual) {
+            this.setState({
+                shippingAddress: {
+                    firstname: billingFirstName,
+                    lastname: billingLastName
+                }
+            });
+        }
+
```

Comment:
If we skip shipping step (only virtual products in cart) then it is impossible to create new account since firstname and last name in createUserOrSaveGuest() function is receiving from shipping step.